scheduler/server/stateful_scheduler.go

```diff
@@ -123,9 +129,10 @@ type averageDuration struct {
 	duration time.Duration
 }
 
-func (ad *averageDuration) update(d time.Duration) {
+func (ad *averageDuration) update(d time.Duration) *averageDuration {
 	ad.count++
 	ad.duration = ad.duration + time.Duration(int64(d-ad.duration)/ad.count)
+	return ad
```

Comment:
I don't see this used anywhere,  maybe remove?

Reply:
thanks, added for previous iteration, not needed anymore